ucsdscheduleme/Repo/ScheduleRepo.cs

```diff
@@ -31,98 +27,105 @@ public class ScheduleRepo
+        /// is popped, it will push all its children onto the stack. Once a node from the last
+        /// layer is popped, then this is a unique schedule. This schedule is checked
+        /// for conflicts and then added if there are no time conflicts.
         /// </summary>
         /// <param name="root">Node to do DFS from</param>
         /// <param name="numClasses">Total number of classes to schedule</param>
         /// <param name="possibleSchedules">Modified list of possible schedules</param>
-        private void DFS(Node root, int numClasses, int numNodes, ref List<List<Section>> possibleSchedules)
+        private void DFS(Node section, int numClasses, ref List<List<Section>> possibleSchedules)
         {
-            Node curr;
-            Stack<Node> stack = new Stack<Node>();
-            bool[] isNodeVisited = new bool[numNodes];
-            List<Section> temp = new List<Section>
-            {
-                root.Section
-            };
-            stack.Push(root);
-            isNodeVisited[0] = true;
+            // start stack with the root
+            Stack<Node> s = new Stack<Node>();
+            s.Push(section);
 
-            while (stack.Count != 0)
+            Section[] tempSchedule = new Section[numClasses];
+
+            // Adds the first class in the correct order on the temp schedule, then push children.
+            while (s.Count != 0)
             {
-                curr = stack.Peek();
-                stack.Pop();
-                temp.Remove(curr.Section);
+                Node curr = s.Pop();
+                tempSchedule[curr.Level] = curr.Section;
 
-                foreach (Node i in curr.Edges)
+                // The popped node was part of the last layer.
+                if (curr.Level == (numClasses - 1))
                 {
-                    if (!isNodeVisited[i.Index] && !Conflict(i.Section, temp))
+                    // If there are no conflicts, copy into list and add to possible.
+                    if (!Conflict(tempSchedule))
                     {
-                        stack.Push(i);
-                        temp.Add(i.Section);
-                        isNodeVisited[i.Index] = true;
+                        List<Section> currSchedule = new List<Section>();
 
-                        if (temp.Count() == numClasses)
+                        for (int i = 0; i < tempSchedule.Count(); i++)
                         {
-                            List<Section> possibleSchedule = new List<Section>();
-                            foreach (Section j in temp)
-                            {
-                                possibleSchedule.Add(j);
-                            }
-                            possibleSchedules.Add(possibleSchedule);
+                            currSchedule.Add(tempSchedule[i]);
                         }
+                        possibleSchedules.Add(currSchedule);
```

Comment:
why we need to copy tempSchedule into the currSchedule and add currSchedule into the possibleSchedules instead of adding tempSchedule directly into the possibleSchedule?

Reply:
because temp schedule is an array and the return type was a list of list of sections. temp schedule is an array because when i tried to make it a list it wouldn’t allow me to use elementAt to change a specific element in the list